### PR TITLE
add check on template extension type - throw error if invalid

### PR DESCRIPTION
**Pull request by eoinmcafee00**
An issue has occurred where users have saved unsupported template file types.
Therefore I have added a check when a user uploads a template file, it has to be yaml, starlark or jsonnet.

### CODE REVIEW

File: plugin/converter/testdata/yaml.template.invalid.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+kind: template
```
Comment (tphoney) [inline plugin/converter/testdata/yaml.template.invalid.yml:1]: should this not be a txt file ?

**Review by tphoney — CHANGES_REQUESTED** [review]

File: plugin/converter/testdata/yaml.template.invalid.yml (hunk 1 above)
Comment (eoinmcafee00) [reply plugin/converter/testdata/yaml.template.invalid.yml:1]: It doesn't really matter as i'm testing the converter logic at this stage, the file extension name check would happen in the handler. I only care about testing the bit of logic https://github.com/drone/drone/pull/3128/files#diff-53a3b4e02181d9a28742c3e46d4120505787eb61138ce0fade533705d460d0deR88 and with the change I made in the handler, should never get to the point where the db has an invalid template

File: plugin/converter/template.go [hunk 2]
```diff
@@ -39,6 +39,7 @@ var (
 	TemplateFileRE          = regexp.MustCompile("^kind:\\s+template+\\n")
 	ErrTemplateNotFound     = errors.New("template converter: template name given not found")
 	ErrTemplateSyntaxErrors = errors.New("template converter: there is a problem with the yaml file provided")
+	errTemplateExtensionInvalid = errors.New("template extension invalid. must be yaml, starlark or jsonnet")
```
Comment (bradrydzewski) [inline plugin/converter/template.go:42]: does this need gofmt ?

Comment (eoinmcafee00) [reply plugin/converter/template.go:42]: ah good catch!

Comment (eoinmcafee00) [reply plugin/converter/template.go:42]: updated.

File: plugin/converter/template.go [hunk 3]
```diff
@@ -36,9 +36,10 @@ import (
 
 var (
 	// TemplateFileRE regex to verifying kind is template.
-	TemplateFileRE          = regexp.MustCompile("^kind:\\s+template+\\n")
-	ErrTemplateNotFound     = errors.New("template converter: template name given not found")
-	ErrTemplateSyntaxErrors = errors.New("template converter: there is a problem with the yaml file provided")
+	TemplateFileRE              = regexp.MustCompile("^kind:\\s+template+\\n")
+	ErrTemplateNotFound         = errors.New("template converter: template name given not found")
+	ErrTemplateSyntaxErrors     = errors.New("template converter: there is a problem with the yaml file provided")
```
Comment (tphoney) [inline plugin/converter/template.go:41]: are all these errors local, they should start with a lowercase letter

**Pull request merged**